pkg/module/filter.go

```diff
@@ -65,13 +64,17 @@ func (f *Filter) AddRule(path string, rule FilterRule) {
 
 // ShouldProcess evaluates path and determines if module should be communicated or not
 func (f *Filter) ShouldProcess(path string) bool {
```

Comment:
this i believe is not used anywhere. please remove if so

Reply:
with 3 states it is not so easy to tell if processing should or shouldn't be done. let's make caller do the call.

Reply:
It is used (i.e. cache_miss_fetcher and cache_miss_reporter) but it's mostly dead code.

Reply:
we should not care about the dead code. we surely dont need cachemiss reported, this an go away, cache miss fetcher we might want to reuse at some point so i would just comment it out with a TODO that it needs to reflect new Rules logic. also it should not be needed there at all.
so action steps: 
- remove cache miss reporter
- remove check from cache_miss_fetcher

Reply:
#461

Reply:
@fedepaol now that #461 is merged, can you create an issue for removing this method?

I'd like to get this PR merged sooner rather than later, so if we have that issue, I'm cool with merging.

Reply:
Issue to follow-up with: https://github.com/gomods/athens/issues/472